lib/bandit/http2/connection.ex

```diff
@@ -513,4 +529,40 @@ defmodule Bandit.HTTP2.Connection do
   defp send_frame(frame, socket, connection) do
     Socket.send(socket, Frame.serialize(frame, connection.remote_settings.max_frame_size))
   end
+
+  defp handle_initial_request({method, request_target, headers, body}, socket, connection) do
+    {_, _, _, path} = request_target
+    headers = [{":scheme", "http"}, {":method", method}, {":path", path} | headers]
+
+    with {:ok, stream} <- StreamCollection.get_stream(connection.streams, 1),
+         true <- accept_stream?(connection),
+         true <- accept_headers?(headers, connection.opts, stream),
+         {:ok, stream} <-
+           Stream.recv_headers(
+             stream,
+             connection.transport_info,
+             connection.telemetry_span,
+             headers,
+             true,
+             connection.plug,
+             body,
```

Comment:
I'm pretty sure we don't need all the `premature_body` plumbing here - you should be able to just doctor up a data frame with the body and call `handle_frame(%Data{})`. There's a fair bit of window space accounting in there. 

Mind you, the RFC doesn't mention anything about whether or not the initial body should be counted against any of the windows. Regardless, I think it's possible for us to do that and remove a whole bunch of the diffs here against stream & adapter (I don't think either of those files should need to change at all).

Reply:
Yep you are right there were no change required for adapter and stream. I have addressed this in 23b526220ee8a44d9e46d6b3b58df2d3785b2408